mitmproxy-rs/src/dns_resolver.rs

```diff
@@ -0,0 +1,19 @@
+use hickory_resolver::config::{LookupIpStrategy, ResolverConfig, ResolverOpts};
+use hickory_resolver::TokioAsyncResolver;
+use pyo3::prelude::{pyfunction, PyResult, Python};
+use pyo3::types::PyAny;
+
+#[pyfunction]
+#[pyo3(signature = (host, is_ipv6))]
+pub fn getaddrinfo(py: Python<'_>, host: String, is_ipv6: bool) -> PyResult<&PyAny> {
```

Comment:
We have a bit of a mismatch here: `getaddrinfo` (with no flags) returns both IPv4 and IPv6 addresses. This method right now only supports either IPv4 or IPv6, but not both at  the same time. We need that as well for use in https://github.com/mitmproxy/mitmproxy/blob/27f990203a6db31b5b95d703e9d7f4fb6380183b/mitmproxy/proxy/server.py#L181. Let's add a `family` int argument instead of the `is_ipv6` bool that can be `0` to support both?